frame/contracts/src/wasm/runtime.rs

```diff
@@ -33,31 +35,67 @@ use sp_io::hashing::{
 	sha2_256,
 };
 
-/// The value returned from ext_call and ext_instantiate contract external functions if the call or
-/// instantiation traps. This value is chosen as if the execution does not trap, the return value
-/// will always be an 8-bit integer, so 0x0100 is the smallest value that could not be returned.
-const TRAP_RETURN_CODE: u32 = 0x0100;
+/// Every error that can be returned from a runtime API call.
+#[repr(u32)]
+pub enum ReturnCode {
```

Comment:
This is only about a contract call to a contract, right? Let's reword that rustdoc to make it convey that fact better.

E.g. "every error that can be returned from `ext_call`/`ext_instantiate`"?

Reply:
No there is also the `KeyNotFound` error that is returned from storage access. Most other calls just trap on error right now. But there might be more errors in the future from additional API.